src/main/java/se/liu/ida/hefquin/cli/modules/ModFederation.java

```diff
@@ -89,147 +88,151 @@ public FederationCatalog getFederationCatalog() {
 		final Model fd = RDFDataMgr.loadModel(filename);
 
-		final ResIterator itSPARQL = fd.listResourcesWithProperty(RDF.type, FD.SPARQLEndpointInterface);
-		while ( itSPARQL.hasNext() ) {
-			final Resource iface = itSPARQL.next();
-			final RDFNode addr = fd.getRequiredProperty(iface, FD.endpointAddress).getObject();
-
-			final String addrStr;
-			if ( addr.isLiteral() ) {
-				addrStr = addr.asLiteral().getLexicalForm();
-			}
-			else if ( addr.isURIResource() ) {
-				addrStr = addr.asResource().getURI();
-			}
-			else {
-				throw new IllegalArgumentException();
-			}
-
-			addSPARQLEndpoint(addrStr);
-		}
-
-		final ResIterator itTPF = fd.listResourcesWithProperty(RDF.type, FD.TPFInterface);
-		while ( itTPF.hasNext() ) {
-			final Resource iface = itTPF.next();
-			final RDFNode addr = fd.getRequiredProperty(iface, FD.exampleFragmentAddress).getObject();
-
-			final String addrStr;
-			if ( addr.isLiteral() ) {
-				addrStr = addr.asLiteral().getLexicalForm();
-			}
-			else if ( addr.isURIResource() ) {
-				addrStr = addr.asResource().getURI();
-			}
-			else {
-				throw new IllegalArgumentException();
-			}
-
-			addTPFServer(addrStr);
-		}
-
-		final ResIterator itBRTPF = fd.listResourcesWithProperty(RDF.type, FD.brTPFInterface);
-		while ( itBRTPF.hasNext() ) {
-			final Resource iface = itBRTPF.next();
-			final RDFNode addr = fd.getRequiredProperty(iface, FD.exampleFragmentAddress).getObject();
-
-			final String addrStr;
-			if ( addr.isLiteral() ) {
-				addrStr = addr.asLiteral().getLexicalForm();
-			}
-			else if ( addr.isURIResource() ) {
-				addrStr = addr.asResource().getURI();
-			}
-			else {
-				throw new IllegalArgumentException();
+		final ResIterator isFedMember = fd.listResourcesWithProperty(RDF.type, FD.FederationMember);
+		// Iterate over all federation members
+		while ( isFedMember.hasNext() ) {
+			final Resource fedMember = isFedMember.next();
+			VocabularyMapping vocabMap = null;
+			if( fd.contains(fedMember, FD.vocabularyMappingsFile) ){
+				final RDFNode pathToMappingFile = fd.getRequiredProperty(fedMember, FD.vocabularyMappingsFile).getObject();
+
+				final String path = pathToMappingFile.toString();
+				if ( verifyValidVocabMappingFile( path ) ) {
+					vocabMap = new VocabularyMappingImpl(path);
+				}
 			}
 
-			addBRTPFServer(addrStr);
-		}
-
-		final ResIterator itNeo4j = fd.listResourcesWithProperty(RDF.type, FD.BoltInterface);
-		while ( itNeo4j.hasNext() ) {
-			final Resource iface = itNeo4j.next();
-			final RDFNode addr = fd.getRequiredProperty(iface, FD.endpointAddress).getObject();
-
-			final String addrStr;
-			if ( addr.isLiteral() ) {
-				addrStr = addr.asLiteral().getLexicalForm();
-			}
-			else if ( addr.isURIResource() ) {
-				addrStr = addr.asResource().getURI();
+			final Resource iface = fedMember.getProperty(FD.interface_).getResource();
+			// Check the type of interface
+			if ( fd.getRequiredProperty(iface, RDF.type).getObject().equals(FD.SPARQLEndpointInterface) ){
```

Comment:
Instead of repeating `fd.getRequiredProperty(iface, RDF.type).getObject()` in all of these if statements, create a program variable for it. That will improve the readability of the code.